test/e2e/framework.go

```diff
@@ -3220,3 +3221,74 @@ func (data *TestData) GetPodLogs(ctx context.Context, namespace, name, container
 	}
 	return b.String(), nil
 }
+
+func (data *TestData) runDNSQuery(
+	podName string,
+	containerName string,
+	podNamespace string,
+	dstAddr string,
+	useTCP bool,
+	dnsServiceIP string) (net.IP, error) {
+
+	digCmd := fmt.Sprintf("dig "+"@"+dnsServiceIP+" +short %s", dstAddr)
+	if useTCP {
+		digCmd += " +tcp"
+	}
+	cmd := []string{
+		"/bin/sh",
+		"-c",
+		digCmd,
+	}
+	fmt.Printf("Running: kubectl exec %s -c %s -n %s -- %s", podName, containerName, podNamespace, strings.Join(cmd, " "))
+	stdout, stderr, err := data.RunCommandFromPod(podNamespace, podName, containerName, cmd)
+	if err != nil {
+		return nil, fmt.Errorf("error when running dig command in Pod '%s': %v - stdout: %s - stderr: %s", podName, err, stdout, stderr)
+	}
+
+	ipAddress := net.ParseIP(strings.TrimSpace(stdout))
+	if ipAddress != nil {
+		return ipAddress, nil
+	} else {
+		return nil, fmt.Errorf("invalid IP address found %v", stdout)
+	}
+}
+
+// patchPodAnnotation Patches an annotation for a given pod in a given namespace. If no annotation is
+// provided as a parameter then it generates a random string value for a predefined key and adds it as annotation.
+// The patchType used is MergePatchType since we intend to make updates to annotations in pods instead of updating fields in complex k8s
+// resources like deployments.
+func (data *TestData) patchPodAnnotation(namespace, podName string, annotation map[string]string) error {
```

Comment:
Usage of this function is weird to me. I don't see the point of the `annotation` map, and you always provide `nil` anyway when you invoke this function. I would suggest this function signature instead:
```go
func (data *TestData) setPodAnnotationToRandomValue(namespace, podName, annotationKey string) error
```

Reply:
Got it. I think i had wanted to do two things here , one was to add an annotation with a random string value as per current use case and second was to cover future use cases wherein instead of adding a random valued annotation we might want to Patch with some custom key,values . 

Have made the necessary change and also added another function. Please review.

Reply:
I think using something like `func (data *TestData) setPodAnnotation(namespace, podName, annotationKey, annotationValue string) error` also makes sense (you can call `randSeq` from the caller), if you prefer. It's probably an improvement.

BTW, you have added `setPodAnnotationToRandomValue` but you haven't removed this function (hence the error from golangci-lint).